tangelo/linq/circuit.py

```diff
@@ -162,6 +165,19 @@ def success_probabilities(self):
         else:
             return self._probabilities
 
+    @property
+    def applied_gates(self):
+        """Returns the list of gates applied during the latest simulation of the circuit.
+
+        If a CMEASURE gate is applied, the resulting circuit can be different from the _gates,
+        The CMEASURE or MEASURE gate will store its measurement result as the parameter.
+
+        Example: circuit = Circuit([Gate("H", 0), Gate("CMEASURE", 0, {"0": Gate("X", 0) "1": []})])
+        will have circuit.applied_gates = [Gate("H", 0), Gate("CMEASURE", 0, parameter="0"), Gate("X", 1)]  or
+                  circuit.applied_gates = [Gate("H", 0), Gate("CMEASURE", 0, parameter="1")]
+        """
+        return self._applied_gates if self.counts.get("CMEASURE", 0) else self._gates
```

Comment:
Please use `'CMEASURE' in self.counts` as this is truly what you want here, rather than having a convoluted way to look for the actual count, returning 0 if you cant find it, and using that as a true/false value to signify that `CMEASURE` appeared in the counts.